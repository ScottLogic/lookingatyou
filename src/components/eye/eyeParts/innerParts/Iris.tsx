import React from 'react';
import tinycolor from 'tinycolor2';
import { IAnimationFrame } from '../../../../utils/pose/animations';
import { IInnerPartProps } from './IInnerPartsProps';

export interface IIrisProps extends IInnerPartProps {
    irisRadius: number;
    animation: IAnimationFrame;
    innerPath: any;
}

const Iris = React.memo((props: IIrisProps) => {
    return (
        <g {...props.groupProps}>
            <circle
                className={'iris'}
                r={props.irisRadius}
                fill={'url(#irisGradient)'}
            />
            <path
                className="irisStyling"
                d={props.innerPath}
                fill={tinycolor(props.animation.irisColor)
                    .darken(10)
                    .toHexString()}
            />
        </g>
    );
});
export default Iris;
