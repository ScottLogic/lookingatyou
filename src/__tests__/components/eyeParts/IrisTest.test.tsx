import { shallow } from 'enzyme';
import React from 'react';
import Iris, {
    IIrisProps,
} from '../../../components/eye/eyeParts/innerParts/Iris';

let props: IIrisProps;
const data = new Uint8ClampedArray(400);
const animation = {
    target: { x: 0, y: 0 },
    dilation: 1,
    openCoefficient: 1,
    irisColor: 'blue',
    duration: 100,
};

for (let i = 0; i < 400; i += 4) {
    data[i + 0] = 190; // R value
    data[i + 1] = 0; // G value
    data[i + 2] = 210; // B value
    data[i + 3] = 255; // A value
}

const imageData = { data, width: 10, height: 10 };

describe('Iris', () => {
    beforeEach(() => {
        props = {
            height: 800,
            width: 1000,
            scleraRadius: 500,
            irisRadius: 200,
            pupilRadius: 100,
            animation,
            reflection: imageData,
            innerPath: '',
            skewTransform: '',
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(<Iris {...props} />).debug();
        expect(wrapper).toMatchSnapshot();
    });
});
