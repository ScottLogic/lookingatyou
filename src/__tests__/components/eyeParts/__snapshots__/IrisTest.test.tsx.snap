// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Iris should render correctly 1`] = `
"<g>
  <circle className=\\"iris\\" r={0.4} fill=\\"url(#irisGradient)\\" />
  <path className=\\"irisStyling\\" d=\\"\\" fill=\\"#0000cc\\" />
</g>"
`;
